SingularityUI/app/selectors/requests/filterSelector.es6

```diff
@@ -74,9 +74,16 @@ export default createSelector([getRequests, getFilter, getUtilizations], (reques
       });
       filteredRequests = _.union(res1, res2).reverse();
     } else {
-      _.each(filteredRequests, (requestParent) => {requestParent.id = id.extract(requestParent);});
+      _.each(filteredRequests, (requestParent) => (requestParent.id = id.extract(requestParent)));
+      // Allow searching by the first letter of each word by applying same
+      // search heuristics to just the upper case characters of each option
+      const options = {
+        extract: Utils.isAllUpperCase(filter.searchFilter)
+          ? (requestParent) => Utils.getUpperCaseCharacters(requestParent.id)
```

Comment:
Doesn't the exact method have to be called in the all uppercase case too?

Reply:
This _is_ the extract method. In the uppercase case, it's returning only the capitals of each request.